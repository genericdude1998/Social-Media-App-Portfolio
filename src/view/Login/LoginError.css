*{
    box-sizing: border-box;
    padding: 0px; 
}

.baseError{
    text-align: center;
    color: red;
    font-weight: bold;
    font-size: 10px;
    margin: 10px;
}