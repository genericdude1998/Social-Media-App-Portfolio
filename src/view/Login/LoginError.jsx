import React from 'react';
import PropTypes from 'prop-types';

const LoginError = ({error}) => {
    return (
        <div>
            {error ? error.errorMessage: null}
        </div>
    );
}

LoginError.propTypes = {
    error: PropTypes.object,
}

export default LoginError;
