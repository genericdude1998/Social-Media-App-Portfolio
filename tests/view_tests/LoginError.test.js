import React from 'react';
import { shallow } from 'enzyme';
import LoginError from '../../src/view/Login/LoginError';
import {mockError} from '../mockValues';

it('should render an error message when given an error', () => {
    let loginErrorWrapper = shallow(
        <LoginError
            error={mockError.errorMessage}
        />
    );
    expect(loginErrorWrapper).toMatchSnapshot();
});
it('should render nothing when no error is given', () => {
    let loginErrorWrapper = shallow(
        <LoginError/>
    );
    expect(loginErrorWrapper).toMatchSnapshot();
});