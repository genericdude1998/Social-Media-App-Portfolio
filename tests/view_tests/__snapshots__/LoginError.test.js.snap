// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render an error message when given an error 1`] = `
<div>
  Mock Error Message!
</div>
`;

exports[`should render nothing when no error is given 1`] = `<div />`;
